openlibrary/plugins/upstream/utils.py

```diff
@@ -1158,6 +1170,56 @@ def reformat_html(html_str: str, max_length: int | None = None) -> str:
         return ''.join(content).strip().replace('\n', '<br>')
 
 
+def get_isbn_10_and_13(isbns: str | list[str]) -> Isbns:
+    """
+    Put a list of ISBNs, or a string of a single ISBN, into the
+    `Isbns` datclass.
+
+    Internet Archive stores ISBNs in a list of strings, with
+    no differentiation between ISBN 10 and ISBN 13. Open Library
+    records need ISBNs in `isbn_10` and `isbn_13` fields.
+
+    >>> isbns = get_isbn_10_and_13(["1576079457", "9781576079454", "1576079392"])
+    >>> isbns.isbn_10
+    ["1576079392", "1576079457"]
+    >>> isbns.isbn_13
+    ["9781576079454"]
+
+    Notes:
+        - this does no validation whatsoever--it merely checks length.
+        - this assumes the ISBNS has no hyphens, etc.
+    """
+    isbn_10 = []
+    isbn_13 = []
+
+    # String of just one ISBN.
+    if isinstance(isbns, str):
+        isbns = isbns.strip()
+        match len(isbns):
+            case 10:
+                isbn_10.append(isbns)
+            case 13:
+                isbn_13.append(isbns)
+
+        return Isbns(isbn_10=isbn_10, isbn_13=isbn_13)
+
+    # List of multiple ISBNs.
+    for isbn in isbns:
+        isbn = isbn.strip()
+        match len(isbn):
+            case 10:
+                isbn_10.append(isbn)
+            case 13:
+                isbn_13.append(isbn)
+
+    if isbn_10:
+        isbn_10.sort()
+    if isbn_13:
+        isbn_13.sort()
```

Comment:
These shouldn't need to be sorted; having them in the order they were in on the record is probably preferable.
```suggestion
```

Reply:
I think my desire to have everything sorted gets the better of me sometimes. :)